addon/mixins/data-adapter-mixin.js

```diff
@@ -67,21 +101,24 @@ export default Ember.Mixin.create({
     @method ajaxOptions
     @protected
   */
-  ajaxOptions() {
+  ajaxOptions(url, type, options) {
     const authorizer = this.get('authorizer');
     Ember.assert("You're using the DataAdapterMixin without specifying an authorizer. Please add `authorizer: 'authorizer:application'` to your adapter.", Ember.isPresent(authorizer));
 
-    let hash = this._super(...arguments);
+    options = options || {};
+    const mixinBeforeSend = options.beforeSend;
+    let hash = this._super(url, type, options);
     let { beforeSend } = hash;
 
     hash.beforeSend = (xhr) => {
-      this.get('session').authorize(authorizer, (headerName, headerValue) => {
-        xhr.setRequestHeader(headerName, headerValue);
-      });
+      if (mixinBeforeSend) {
+        mixinBeforeSend(xhr);
+      }
       if (beforeSend) {
         beforeSend(xhr);
       }
     };
+
     return hash;
   },
```

Comment:
We need it because REST adapter overwrites `beforeSend` in its own `ajaxOptions`, it doesn't check if `beforeSend` has already been assigned. We re-assign it here.